rust-runtime/aws-smithy-types/src/error/interceptor.rs

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Errors related to smithy interceptors
+
+use std::fmt;
+
+/// A generic error that behaves itself in async contexts
+pub type DynError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+/// Errors related to smithy interceptors.
+#[derive(Debug)]
+pub enum InterceptorError {
```

Comment:
Is there a reason for this to be an actionable error instead of an informative error? (per [RFC-22](https://github.com/awslabs/smithy-rs/blob/main/design/src/rfcs/rfc0022_error_context_and_compatibility.md))

Also needs to implement `Error::source`

Reply:
You're quite right. Good callout.